dandiapi/api/views/serializers.py

```diff
@@ -78,6 +81,17 @@ def get_is_starred(self, dandiset):
 
 class CreateDandisetQueryParameterSerializer(serializers.Serializer):
     embargo = serializers.BooleanField(required=False, default=False)
+    funding_source = serializers.CharField(required=False, allow_blank=True)
+    award_number = serializers.CharField(required=False, allow_blank=True)
+    embargo_end_date = serializers.DateField(
+        default=lambda: timezone.now().date() + timedelta(days=730)
+    )
+
+    def validate_embargo_end_date(self, value: date) -> date:
+        max_end_date = timezone.now().date() + timedelta(days=730)
+        if value > max_end_date:
+            raise ValidationError('Embargo end date cannot be more than 730 days in the future')
+        return value
```

Comment:
@bendichter how far in the future should we allow embargo end dates? I went with 730 days to match the default, but let me know if there's a different value we should use.

Reply:
I see this is already set in the frontend as 5 years https://github.com/dandi/dandi-archive/pull/2462/files#diff-5c8140317c91f14905bc0bb134aeadbb668ec81c2b341dad8ff27e2c0a8e061aR299-R302 - I'll update the backend logic accordingly